tests/src/python/test_qgsattributeform.py

```diff
@@ -163,7 +163,7 @@ def test_on_update(self):
 
         layer = QgsVectorLayer("Point?field=age:int", "vl", "memory")
         # set default value for numbers to [1, {age}], it will depend on the field age and should update
-        field = QgsField('numbers', QVariant.List, 'array')
+        field = QgsField('numbers', QVariant.List, subType=QVariant.Int)
```

Comment:
@signedav , is that change necessary? If so, can you explain? If not, let's revert.

Reply:
Before on form.save the values are gone. In this test we dont do it so it doesnt matter, but in the next one we do. But if someone would extend this function they might run into the same problem when this subtype is still array.

Reply:
Okay, I change it back to avoid confusion because this PR did not change anything in behavior of that test.